src/gc.rs

```diff
@@ -100,101 +196,314 @@ impl<'a, D> DatabaseTrieMut<'a, D> {
     }
 }
 
+#[derive(Default)]
+pub struct MapWithCounter {
+    counter: DashMap<H256, usize>,
+    data: DashMap<H256, Vec<u8>>,
+}
+impl MapWithCounter {
+    fn increase(&self, key: H256) -> usize {
+        self.counter
+            .entry(key)
+            .and_modify(|count| {
+                *count += 1;
+            })
+            .or_insert(1);
+        trace!("{} count++ is {}", key, *self.counter.get(&key).unwrap());
+        *self.counter.get(&key).unwrap()
```

Comment:
`Entry::or_insert` returns `RefMut`. So we can avoid next two `get` calls just by dereferencing it.